src/python/pants/core/util_rules/posix_tools.py

```diff
@@ -0,0 +1,50 @@
+# Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+
+from pants.engine.internals.selectors import MultiGet
+from pants.engine.process import BinaryPath, BinaryPathRequest, BinaryPaths, BinaryPathTest
+from pants.engine.rules import Get, collect_rules, rule
+from pants.util.logging import LogLevel
+
+
+@dataclass(frozen=True)
+class PosixTools:
+    """ `BinaryPath`s for standard Unix tools that may be used in internal shell scripts."""
+    ln: BinaryPath
```

Comment:
I think that unless we were going to optimize this lookup to only fork one discovery process (which seems challenging?), coupling together a bunch of tools like this will mean that we end up discovering a lot more tools than people use in practice. Having separate things to request in your @rule is a better for laziness.

Reply:
Indeed: https://github.com/pantsbuild/pants/pull/12947

Reply:
Fine